NewForce-Cohort-8/tabloidfullstack-the-cage-crushers
# Subscription Feed

Description

Subscription Feed

When a user visits the homepage and has subscriptions, they should encounter a list of posts authored by those they are subscribed to.

Fixes # (issue)

[Trello Ticket 31](https://trello.com/c/EtLHQKJk/31-31-view-subscribed-posts-on-homepage)

Type of change

New feature (non-breaking change which adds functionality)

Testing Instructions


1. git checkout main
2. git fetch --a
3. git checkout ` ht-subscribelist`
4. If you don't see the most recent version of my files, git pull origin ` ht-subscribelist`
5. If you haven't created the Tabloid Full Stack database, please do so.
6. Use this to fill out tables and columns: [SQL data](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-cage-crushers/blob/main/SQL/01_Tabloid_Create_DB.sql)
7. Use this to seed the database: [SQL Seed Data](https://github.com/NewForce-Cohort-8/tabloidfullstack-the-cage-crushers/blob/main/SQL/02_Tabloid_Seed_Data.sql)
8. Click the execute with debugger solid green button to launch the app.
9. Then, run the application in a browser window by running `npm start` in the client/tabloid folder. You should see the 'Home' page with a navigation bar at the top of the page. If not already logged out, log out (with the link in the navbar), and 'Log in' with the following credentials.
10. Email: [foo@bar.com](mailto:foo@bar.com)
11. Ensure the user has a subscription by clicking on "Posts" and navigating to the Post Details page of a post by click on a specific post title.
12. Click the "subscribe" button to create a subscription to that author's post. Note: If you are already subscribed a modal will pop up to inform you that you are already subscribed.  Also, the unsubscribe feature is not implemented at this time therefore you cannot unsubscribe from that author.
13. Redirect back to the "HOME" page and you should see a list of posts from the author that you previously subscribed to. 

 
Checklist:


- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- My changes generate no new warnings or errors
- I have added test instructions that prove my fix is effective or that my feature works
